pkg/controller/render/render_controller.go

```diff
@@ -636,3 +724,521 @@ func getMachineConfigsForPool(pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.
+	}
+
+	// The the trust for the repo serving cert is not in our system trust by default, but we know what signed it, so we
+	// get the pubkey for that and add it to our tls config so we trust it
+	serviceServingSignerCA, err := ctrl.kubeclient.CoreV1().ConfigMaps("openshift-config").Get(context.TODO(), "openshift-service-ca.crt", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve the openshift-service-ca.crt configmap: %w", err)
+	}
+
+	// TODO(jkyros): This is better than using skipverify/"nosec" to get past verify, but figure out error cases here later
+	rootCAs := x509.NewCertPool()
+	rootCAs.AppendCertsFromPEM([]byte(serviceServingSignerCA.Data["service-ca.crt"]))
+
+	// Add the service serving CA to our trust bundle in our tls config
+	tr := &http.Transport{TLSClientConfig: &tls.Config{
+		RootCAs: rootCAs,
+	}}
+
+	crane.WithTransport(tr)
+
+	// Retrieve our service account token (I know there is probably a more kubernetes-y way to do this, but this works for now)
+	stoken, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	if err != nil {
+		return fmt.Errorf("Unable to get service token: %s", stoken)
+	}
+
+	// Create our auth object to let crain/containerregistry auth to the integrated repo
+	au := &authn.Basic{
+		Username: "machine-config-controller",
+		Password: string(stoken),
+	}
+
+	// Push our image into the integrated repo
+	if err := crane.Push(newImg, tag.String(), crane.WithTransport(tr), crane.WithAuth(au)); err != nil {
+		return fmt.Errorf("Failed to push image %s: %s", tag.String(), err)
+	}
+	glog.Infof("Image %s successfully pushed.", tag.String())
+
+	// TODO(jkyros): I wonder if I can use an EnvVarSource here and reference something without having to stuff this in
+	var whichConfig = corev1.EnvVar{
+		Name:  "machineconfig",
+		Value: generated.Name,
+	}
+
+	// Create a buildconfigrequest to trigger the build (instead of using the imagestream trigger), because if we let the
+	// imagestream do it on its own, we can't stuff in the metadata (like which rendered-config it is equivalent to).
+	// It looks like the imagetag object that comes out will let me see the labels and the env, so I could just stuff it
+	// in "env" but that seems hacky
+
+	// TODO(jkyros): If we skip the buildconfig (or make the build from the buildconfig ourselves) and just make this a build, then we
+	// can include whatever dynamic labels/metadata we want
+	br := &buildv1.BuildRequest{
+		//TypeMeta:    metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("mco-build-content-%s", pool.Name)},
+		//Env: []corev1.EnvVar{whichConfig},
+		TriggeredBy: []buildv1.BuildTriggerCause{
+			{Message: "The machine config controller"},
+		},
+		DockerStrategyOptions: &buildv1.DockerStrategyOptions{
+			BuildArgs: []corev1.EnvVar{whichConfig},
+			//NoCache:   new(bool),
+		},
+	}
+
+	// Trigger our build manually, we don't have to wait, the pool will figure it out when it's available
```

Comment:
```suggestion
	// Trigger our build manually. We don't have to wait, the pool will figure it out when it's available
```

Reply:
This one is my favorite, my grammar is also bad! :smile: